src/grep.rs

```diff
@@ -116,7 +121,7 @@ fn grep_bytes(
     }
   }
   else {
-    let mut matches = pattern.find_iter(buffer);
+    let matches = &mut pattern.find_iter(buffer);
```

Comment:
I believe a reference has no benefit here. If I'm right, could you revert this please?